graylog2-web-interface/src/components/sidecars/configuration-forms/ConfigurationForm.tsx

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import PropTypes from 'prop-types';
+import React, { useState, useRef, useEffect } from 'react';
+import lodash from 'lodash';
+
+import history from 'util/History';
+import { ColorPickerPopover, FormSubmit, Select, SourceCodeEditor } from 'components/common';
+import { Button, Col, ControlLabel, FormControl, FormGroup, HelpBlock, Row, Input } from 'components/bootstrap';
+import Routes from 'routing/Routes';
+import ColorLabel from 'components/sidecars/common/ColorLabel';
+import { CollectorConfigurationsActions } from 'stores/sidecars/CollectorConfigurationsStore';
+import { CollectorsActions } from 'stores/sidecars/CollectorsStore';
+
+import SourceViewModal from './SourceViewModal';
+import ImportsViewModal from './ImportsViewModal';
+import ConfigurationTagsSelect from './ConfigurationTagsSelect';
+
+import type { Collector, Configuration, ConfigurationSidecarsResponse } from '../types';
+
+type Props = {
+  action: string,
+  configuration: Configuration,
+  configurationSidecars: ConfigurationSidecarsResponse,
+};
+
+const ConfigurationForm = ({
+  action,
+  configuration,
+  configurationSidecars,
+}: Props) => {
+  const initFormData = {
+    id: configuration.id,
+    name: configuration.name,
+    color: configuration.color,
+    collector_id: configuration.collector_id,
+    template: configuration.template || '',
+    tags: configuration.tags || [],
+  };
+
+  const [collectors, setCollectors] = useState<Collector[]>([]);
+  const [formData, setFormData] = useState(initFormData);
+  const [error, setError] = useState(false);
+  const [validationErrors, setValidationErrors] = useState({});
+  const previewModal = useRef(null);
+  const uploadsModal = useRef(null);
+
+  useEffect(() => {
+    CollectorsActions.all().then((response) => setCollectors(response.collectors));
+  }, []);
+
+  const defaultTemplates = {};
+
+  const _isTemplateSet = (template) => {
+    return template !== undefined && template !== '';
+  };
+
+  const _hasErrors = () => {
+    return error || !_isTemplateSet(formData.template);
+  };
+
+  const _validateFormData = (nextFormData, checkForRequiredFields) => {
+    CollectorConfigurationsActions.validate(nextFormData).then((validation) => {
```

Comment:
Somehow the CollectorConfigurationsActions does not have the validate action and the store too. did you omit the files during commit?

Reply:
Nope nothing was omitted there

Reply:
My bad, I was looking to the wrong file in the Plugin Collector!